src/server/api.php

```diff
@@ -1305,124 +1258,127 @@ function get_meta_afhsb() {
   }
   return $epidata;
 }
-function meta_delphi() {
-  $epidata = array();
+function meta_delphi(IRowPrinter &$printer) {
   $query = 'SELECT `system`, min(`epiweek`) `first_week`, max(`epiweek`) `last_week`, count(1) `num_weeks` FROM `forecasts` GROUP BY `system` ORDER BY `system` ASC';
   $fields_string = array('system');
   $fields_int = array('first_week', 'last_week', 'num_weeks');
-  execute_query($query, $epidata, $fields_string, $fields_int, null);
-  return count($epidata) === 0 ? null : $epidata;
+  execute_query($query, $printer, $fields_string, $fields_int, null);
 }
 
-// all responses will have a result field
-$data = array('result' => -1);
-// connect to the database
-if(database_connect()) {
-  // select the data source
-  $source = isset($_REQUEST['source']) ? strtolower($_REQUEST['source']) : null;
+
+/*
+ * main function
+ */
+function main() {
+  $source = isset($_REQUEST['source']) ? strtolower($_REQUEST['source']) : '';
```

Comment:
wasn't there a recent change to replace "source" with "endpoint". (not actual replacement, but making "source" and "endpoint" synonyms.) should `$_REQUEST['endpoint']` appear here too?

Reply:
(it's not merged yet)

Reply:
not it has been merged into a `source-to-endpoint` branch. But I merged that one in